### ISSUE TITLE
Blank Screen on iOS and Safari

### ISSUE THREAD

**Issue by Joewayjose** [DestinyItemManager/DIM#9308]
### DIM Version

Unable to Verify

### Browser Details

Safari, and iOS app

### OS Details

iOS 16.4

### Describe the bug

After authentication, the app is just a blank white screen. The website does the same. It does work correctly on the iOS Chrime browser however

**Comment by bhollis** [comment]
Few things to try:
1. Uninstall the app and reinstall
2. Clear your Safari cache
3. Install the webapp instead of using the App Store version

**Comment by Joewayjose** [comment]
I have done all three. I’ve even moved between a beta iOS  16.4 version and back to the standard.

**Comment by bhollis** [comment]
Interesting. Do you have any content blockers installed? Any particular privacy settings enabled for Safari? Were you opening DIM in a Private Browsing window?

**Comment by mrpunkin** [comment]
I too am having this issue since updating to iOS 16.4

**Comment by mrpunkin** [comment]
Some follow-up on this since I'm a web-developer myself:

Opening the inspector from my iPhone via my Mac shows that it's getting a 200 response via a service worker from the server, but it's zero bytes. There isn't anything showing up in console or assets or anything. The only thing being transferred is the 200 blank response back to my phone.

**Comment by bhollis** [comment]
I'll have to upgrade and try it out myself. I don't use the iOS app, I only have it installer from the web.

**Comment by bhollis** [comment]
I just updated to 16.4 and both my version installed from Safari and the App Store version worked fine with PSN login. I still think there might be some setting affecting you.

**Comment by bhollis** [comment]
If you go to Settings > Safari > Clear Website History And Data, does that help?

**Comment by mrpunkin** [comment]
I can't comment for @Joewayjose, but I use "Screen Time" and have "Limit Adult Websites" enabled. I tried just turning that off as it's literally the only "custom" thing I do with Safari and it started working. Turning it back on kills the site / app. I even added the domains to the always allowed whitelist but if that setting is enabled DIM no longer works.

**Comment by bhollis** [comment]
Well that's exciting. Not sure how we got on Apple's naughty list...

**Comment by mrpunkin** [comment]
I think the odd thing is not being able to whitelist it. Makes me think it's actually a third-party domain such as a cloudfront one or something that's being flagged. I have plenty of sites in my whitelist that work just fine after being added.

**Comment by Joewayjose** [comment]
Yep. Turning on the Adult website filter fixed it. Hilarious. Thank you for the assistance. Should I close this or leave open since DIM will still have this issue with logins as the setting is

**Comment by mrpunkin** [comment]
> Yep. Turning on the Adult website filter fixed it.

Turning it on or turning it off. In my case it fails when I have the filter enabled.

**Comment by mrpunkin** [comment]
Yep, okay, so we are having the same experience then. Seems like it's just a matter of knowing what domains to whitelist for those of us that use the filtering. I added the following to my whitelist but still have issues:

- app.destinyitemmanager.com
- beta.destinyitemmanager.com
- cloudfront.com

**Comment by bhollis** [comment]
We don't use cloudfront. We talk to our own domains plus Bungie.net.

**Comment by mrpunkin** [comment]
Thinking more about this, I don't think it's as simple as DIM having their domain mis-flagged or something, as there isn't the standard screen showing that it'd been blocked. Instead it loads the domain fine but then never loads any of the app code from that point on.

I've whitelisted every domain I can see DIM interacting with on Chrome desktop (including Google's ones for fonts, tag manager, etc.) and still only get a white screen as if the JS or service worker is dying or something.

**Comment by mrpunkin** [comment]
Still having the same issues. I also upgraded to iOS 16.4.1 and still am having these issues. If I remove all data associated with bungie.net and destinyitemmanager.com it'll load, auth, and sometimes load inventory for a few moments but then goes back to a white screen on refresh.

Could it be an issue with DIM Sync?

**Comment by bhollis** [comment]
All indications are that Apple uses a client side ML model to determine whether a page is Adult or not. Who knows why it's doing so here - too many images, the capitalized word Destiny, et,

**Comment by mrpunkin** [comment]
Wouldn't that be true on initial page load after killing off the service worker too? The fact that it loads fine initially, but subsequent refreshes that rely on the service worker don't tells me it's not actually the adult content filtering that's the root cause, perhaps just exacerbating it.

**Comment by bhollis** [comment]
It's also possible that Apple just broke service workers

**Comment by bhollis** [comment]
Yup looks like it. Classic Safari. 

https://stackoverflow.com/questions/75885199/vue2-pwa-is-not-working-on-ios-16-4-devices

**Comment by mrpunkin** [comment]
Well without disabling content restrictions or changing anything I was able to get it to load and re-load numerous times just fine within safari after unregistering all service workers, but when I add it to my homescreen it exhibits the same white screen behavior. I've noticed that when I do get the white screen that Safari is showing dual service workers for the open page.

In this currently working iteration there is only the one showing up. I'm not discounting Apple having a bug here, as that link clearly shows others having this specific issue as well, however it is also odd that sometimes it works fine and sometimes it doesn't. 

I know, not enough to really go off of, just trying to provide as much detail as I can.
